main.py

```
@@ -0,0 +1,63 @@
+    if input_name.name in tmp_db:
+        raise HTTPException(status_code=400, detail="이름이 이미 존재합니다")
+
+    # 성공!
+    try:
+        NameRepository.add_name(input_name.name)
+        return {"message": "이름이 추가되었습니다", "name": input_name.name}
+    except Exception as e:
+        # 예상치 못한 오류가 발생한 경우
+        raise HTTPException(
+            status_code=500, detail="서버 오류가 발생했습니다 : " + str(e)
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)